### PR TITLE
Ride history page

### PR DESCRIPTION
**Pull request by helenxuyang**
### Summary 

This pull request is the first step towards implementing the ride history page

- [x] implemented displaying past rides by date
- [x] included text wrap solution, but still partially hardcoded 

### Notes 

- text wrapping uses a container of half of the screen width, I couldn't find an ideal solution because of all of the nested layout widgets

### CODE REVIEW

File: lib/Profile.dart
Comment (lsizemore8) [inline lib/Profile.dart]: Is there a specific reason to hardcode this as opposed to putting it in a theme like it was before?

File: lib/Ride.dart
Comment (lsizemore8) [inline lib/Ride.dart]: Nit: Any reason why the indentation was removed?

File: lib/Ride.dart
Comment (lsizemore8) [inline lib/Ride.dart]: Nit: is there a way to make the theme titles more descriptive? Like when would title3 be used? if that's not possible due to changing designs feel free to ignore this

File: lib/Ride.dart
Comment (lsizemore8) [inline lib/Ride.dart]: this one is good with regards to the theme title.

File: lib/Ride.dart
Comment (lsizemore8) [inline lib/Ride.dart]: Same thing with the theme. Does the gray3 correspond to something in the design (in which case my bad, please ignore) or is it just the 3rd gray color? As someone coming into the project I'm not sure what this theme represents.

File: lib/CarriageTheme.dart
Comment (lsizemore8) [inline lib/CarriageTheme.dart]: Nit: For consistency, is there a way to stick with either hex or RGB? A lot of the colors elsewhere in this PR are hex while these are RGBO. Or is that a design decision?

**Review by lsizemore8 — COMMENTED** [review]
I'm a little confused why some of the themes are prebuilt themes and others are hardcoded in. Is it just to keep the PR small/the hardcoded remining here will be fixed as designs get fleshed out. Mainly just caught nits & I'm not going to approve or request changes given I'm still catching up on Dart.

File: lib/Profile.dart
Comment (helenxuyang) [reply lib/Profile.dart]: On the Figma it's a unique color that isn't part of the theme, I now realize it might have been unintentional since I think this page was made before they added the design system, so I'll check on that!

File: lib/Ride.dart
Comment (helenxuyang) [reply lib/Ride.dart]: I think this was an accident sorry! Reminds me, some of our indentation is really messed up from passing around between Android Studio and VSCode which seems to mess up end parentheses--maybe would be a simple code cleanup task?

Comment (lsizemore8) [reply lib/Ride.dart]: Definitely may be a good cleanup task

File: lib/Ride.dart
Comment (lsizemore8) [reply lib/Ride.dart]: Also if title3 or gray3 are the names from the design system, definitely don't change it then.

File: lib/Ride.dart
Comment (lsizemore8) [reply lib/Ride.dart]: Discussed offline, didn't realize this related to the design system. likely okay then

File: lib/CarriageTheme.dart
Comment (helenxuyang) [reply lib/CarriageTheme.dart]: Whoops, this was just from me being lost on Figma. Will change to be consistent!

File: lib/Ride.dart
Comment (helenxuyang) [reply lib/Ride.dart]: Talked about this on Slack but just for the record if anyone else has the same question, the theme titles all correspond to the titles in the design system in Figma where they're called title 1/2/3 and gray 1/2/3!

File: lib/CarriageTheme.dart
Comment (lsizemore8) [reply lib/CarriageTheme.dart]: Discussed offline, changed in a different commit

File: lib/Ride.dart
Comment (lsizemore8) [inline lib/Ride.dart]: Nit: indentation again

**Review by lsizemore8 — APPROVED** [review]
Looks good!

**Review by lsizemore8 — APPROVED** [review]
Spoken offline, approving to unblock authentication, this needs to be tested on different phone types after the merges

**Pull request merged**
